File: Data Structures & Algorithms/majority-element-ii/submission-0.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        std::map<int, int> seen;
        for (int num : nums){
            auto it = find(nums.begin(), nums.end(), num);
            if (it == nums.end()) {
                seen.insert({num, 1});
            }
            else {
                seen[num]++;
            }
        }
        int size = nums.size();
        std::vector<int> ans;
        for (auto [key, val] : seen) {
            if (val > size / 3) {
                ans.push_back(key);
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ans = None
        count = 0
        for val in nums:
            if val == ans:
                count += 1
            else:
                count -= 1
                if count <= 0:
                    ans = val
                    count = 1
        return ans